src/ccache.cpp

```diff
@@ -3359,17 +3359,53 @@ set_up_config(Config& config)
     MTR_END("config", "conf_read_secondary");
 
     if (config.cache_dir().empty()) {
-      fatal("configuration setting \"cache_dir\" must not be the empty string");
+      // If the system cache_dir isn’t set, first check for the legacy path.
+      char* home = getenv("HOME");
+      if (!home) {
+        fatal("HOME must be defined");
+      }
```

Comment:
This won't work on Windows – use `get_home_directory()` instead.

Tangentially, I would also like to improve the default location on macOS (`~/Library/Preferences/ccache` and `~/Library/Caches/ccache`) and Windows as well (`%APPDATA%/ccache/` instead of `%APPDATA%/.ccache`), but that doesn't have to be part of this PR.

Reply:
Done for `get_home_directory()`.  I don’t have any way to test these two platforms, aside from Wine, so I’ll indeed leave that out for some other contributor.

Although I remember reading that on macOS you have to put some plist file in that Preferences path, so you may want to use XDG base directories there too.